OpenRA.Game/Network/Order.cs

```diff
@@ -386,6 +397,21 @@ public byte[] Serialize()
 									w.Write(Target.SerializablePos.X);
 									w.Write(Target.SerializablePos.Y);
 									w.Write(Target.SerializablePos.Z);
+
+									// Don't send extra data over the network that will be restored by the Target ctor
+									var terrainPositions = Target.SerializableTerrainPositions.Length;
+									if (terrainPositions == 1 && Target.SerializableTerrainPositions[0] == Target.SerializablePos)
+										w.Write((byte)0);
```

Comment:
What if the array has length 0? On deserializing we will interpret that as an array with the center position instead.